pkg/kubelet/status/status_manager.go

```diff
@@ -381,6 +381,10 @@ func (m *manager) updateStatusInternal(pod *v1.Pod, status v1.PodStatus, forceUp
 		klog.Errorf("Status update on pod %v/%v aborted: %v", pod.Namespace, pod.Name, err)
 		return false
 	}
+	if err := checkContainerStateTransition(oldStatus.EphemeralContainerStatuses, status.EphemeralContainerStatuses, pod.Spec.RestartPolicy); err != nil {
```

Comment:
Can we amend the existing test coverage to check that ephemeral containers have their state transitions checked?